connectors/amazons3/connector/src/main/java/org/apache/manifoldcf/authorities/authorities/amazons3/AmazonS3Authority.java

```diff
@@ -0,0 +1,599 @@
+
+  protected long lastSessionFetch = -1L;
+
+  protected static final long timeToRelease = 300000L;
+
+  protected AmazonS3 amazonS3;
+
+  protected boolean connected = false;
+
+  protected String amazons3ProxyHost = null;
+
+  protected String amazons3ProxyPort = null;
+
+  protected String amazons3ProxyDomain = null;
+
+  protected String amazons3ProxyUserName = null;
+
+  protected String amazons3ProxyPassword = null;
+
+  protected String amazons3AwsAccessKey = null;
+
+  protected String amazons3AwsSecretKey = null;
+
+  public AmazonS3Authority() {
+
+  }
+
+  @Override
+  public void disconnect() throws ManifoldCFException {
+    amazons3AwsAccessKey = null;
+    amazons3AwsSecretKey = null;
+
+    amazons3ProxyHost = null;
+    amazons3ProxyPort = null;
+    amazons3ProxyDomain = null;
+    amazons3ProxyUserName = null;
+    amazons3ProxyPassword = null;
+  }
+
+  @Override
+  public void connect(ConfigParams configParams) {
+    super.connect(configParams);
+    // aws access and secret keys
+    amazons3AwsAccessKey = configParams
+        .getParameter(AmazonS3Config.AWS_ACCESS_KEY);
+    amazons3AwsSecretKey = configParams
+        .getObfuscatedParameter(AmazonS3Config.AWS_SECRET_KEY);
+
+    // proxy values
+    amazons3ProxyHost = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_HOST);
+    amazons3ProxyPort = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_PORT);
+    amazons3ProxyDomain = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_DOMAIN);
+    amazons3ProxyUserName = configParams
+        .getParameter(AmazonS3Config.AMAZONS3_PROXY_USERNAME);
+    amazons3ProxyPassword = configParams
+        .getObfuscatedParameter(AmazonS3Config.AMAZONS3_PROXY_PASSWORD);
+  }
+
+  /**
+   * Test the connection. Returns a string describing the connection
+   * integrity.
+   *
+   * @return the connection's status as a displayable string.
+   */
+  @Override
+  public String check() throws ManifoldCFException {
+    // connect with amazons3 client
+    Logging.authorityConnectors.info("Checking connection");
+
+    try {
+      // invokes the check thread
+      CheckThread checkThread = new CheckThread(getClient());
+      checkThread.start();
+      checkThread.join();
+      if (checkThread.getException() != null) {
+        Throwable thr = checkThread.getException();
+        return "Check exception: " + thr.getMessage();
+      }
+      return checkThread.getResult();
+    }
+    catch (InterruptedException ex) {
+      Logging.authorityConnectors.error(
+          "Error while checking connection", ex);
+      throw new ManifoldCFException(ex.getMessage(), ex,
+          ManifoldCFException.INTERRUPTED);
+    }
+
+  }
+
+  /**
+   * Get the Amazons3 client, relevant access keys should have been posted
+   * already
+   * @return
+   */
+  protected AmazonS3 getClient() {
+    if (amazonS3 == null) {
+      try {
+        BasicAWSCredentials awsCreds = new BasicAWSCredentials(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).